tools/test.py

```diff
@@ -3,127 +3,120 @@
 # Runs the full test suite.
 # Usage: ./tools/test.py out/Debug
 import os
+import subprocess
 import sys
 import unittest
 
-from integration_tests import integration_tests
-from deno_dir_test import deno_dir_test
-from util import build_path, enable_ansi_colors, executable_suffix, run, rmtree
-from util import run_output, tests_path, green_ok
-from unit_tests import unit_tests
-from util_test import util_test
-from setup_test import setup_test
-from benchmark_test import benchmark_test
-from repl_test import repl_tests
-from fetch_test import fetch_test
-from fmt_test import fmt_test
-import subprocess
-import http_server
-
-
-def check_exists(filename):
-    if not os.path.exists(filename):
-        print "Required target doesn't exist:", filename
-        print "Run ./tools/build.py"
-        sys.exit(1)
-
-
-def test_no_color(deno_exe):
-    sys.stdout.write("no_color test...")
-    sys.stdout.flush()
-    t = os.path.join(tests_path, "no_color.js")
-    output = run_output([deno_exe, "run", t], merge_env={"NO_COLOR": "1"})
-    assert output.strip() == "noColor true"
-    t = os.path.join(tests_path, "no_color.js")
-    output = run_output([deno_exe, "run", t])
-    assert output.strip() == "noColor false"
-    print green_ok()
-
-
-def exec_path_test(deno_exe):
-    cmd = [deno_exe, "run", "tests/exec_path.ts"]
-    output = run_output(cmd)
-    assert deno_exe in output.strip()
+from benchmark_test import TestBenchmark
+from deno_dir_test import TestDenoDir
+from fetch_test import FetchTest
+from fmt_test import FmtTest
+from integration_tests import TestIntegrations
+from repl_test import TestRepl
+from setup_test import TestSetup
+from unit_tests import JsUnitTests
+from util_test import TestUtil
+
+from is_tty_test import TestIsTty
+# NOTE: These tests are skipped on Windows
+from permission_prompt_test import permission_prompt_tests
+from complex_permissions_test import complex_permissions_tests
+
+from http_server import Spawn
+from util import (DenoTestCase, ColorTextTestRunner, enable_ansi_colors,
+                  executable_suffix, run, run_output, rmtree, tests_path,
+                  test_args)
+
+
+class TestLib(DenoTestCase):
```

Comment:
s/TestLib/TestTarget/

Reply:
Should this be split up further? In a separate files?